packages/starknet-types/Cargo.toml

```diff
@@ -0,0 +1,28 @@
+[package]
```

Comment:
are these types only used in ampd? or are they used in contracts as well? If they are only used in ampd, I think they should just live in a folder in ampd, close to where they are used. I see the packages folder as a place for things that are used across the entire repo.

Reply:
The StarknetMessage type is used in the `multisig-prover`.
I can move the events and other types in ampd.

I'll use this thread to ask a question. We plan to migrate to using the `cainome` crate in our relayer. It has an `abigen!()` macro, which generates starknet types based on the ABI JSON file.

Do you prefer us to import that dependency and to maintain an ABI JSON file, or for us to leave the types in this folder? If we do - we'll delete all the types apart from `StarknetMessage` and use the generated ones. Else - we keep the ones here and only update if needed (rare to never).

Reply:
Please check [this commit](https://github.com/axelarnetwork/axelar-amplifier/pull/731/commits/f46f33ab42a3fa5f17417e46e9f65e64d06a8152), in which I moved part of the modules to ampd and renamed `starknet-types` to `starknet-message`, keeping it in `/packages`, because it's used in the prover.

Reply:
yeah the prover code should live in your fork, and I'd prefer not to make changes to this repo that only really make sense for your fork. It makes things confusing. Eventually we will refactor ampd as well, so that starknet specific ampd code can also live in your fork as well.

I actually think I prefer the generated types. Can you try that and we see how it looks?

Reply:
It will require quite a refactor. Is it ok if we leave that for the optimisations phase of the project?

What has to be done to remove the files in the new `/types` folder:
- the [cainome](https://github.com/cartridge-gg/cainome) library will be included as a dependency.
- an ABI json file needs to be kept up to date.
- we pass that file to the `abigen!()` macro.
- it generates **a lot** of stuff in memory, including an RPC client for the contract to which the ABI json corresponds (the gateway contract). In `ampd` we need at most 5% of what it will generate.
- we substitute the `ContractCall` `SignersRotated` and `ByteArray` types with the generated ones.
- remove current implementations and fix tests.